src/framework-configuration.js

```diff
@@ -256,17 +303,37 @@ export class FrameworkConfiguration {
   /**
    * Configures an external, 3rd party plugin before Aurelia starts.
    * @param plugin The ID of the 3rd party plugin to configure.
-   * @param config The configuration for the specified plugin.
+   * @param pluginConfig The configuration for the specified plugin.
    * @return Returns the current FrameworkConfiguration instance.
  */
-  plugin(plugin: string, config?: any): FrameworkConfiguration {
+  plugin(
+    plugin: string | ((frameworkConfig: FrameworkConfiguration) => any) | FrameworkPluginInfo,
+    pluginConfig?: any
+  ): FrameworkConfiguration {
     assertProcessed(this);
 
-    if (typeof (plugin) === 'string') {
-      return this.plugin({ moduleId: plugin, resourcesRelativeTo: [plugin, ''], config: config || {} });
+    let info: FrameworkPluginInfo;
+    switch (typeof plugin) {
+    case 'string':
+      info = { moduleId: plugin, resourcesRelativeTo: [plugin, ''], config: pluginConfig || {} };
+      break;
+    case 'function':
+      if (this.queuedPlugins.indexOf(plugin) !== -1) {
```

Comment:
This doesn't dedup calls between old string-based and new function-based APIs.
Does it matter?

Reply:
Maybe we can still detect which `configure` functions that have been called, so we can avoid duping. doing this probably help remove the code related to de-duping the queue, or maybe it can still be useful for startup perf. To dedup / add interop between string based and function based APIs, there would be a simple queue and check [here](https://github.com/bigopon/aurelia-framework/blob/a9036803e3b686076c76e803cdb5bbf51a8979ed/src/framework-configuration.js#L51) and [here](https://github.com/bigopon/aurelia-framework/blob/a9036803e3b686076c76e803cdb5bbf51a8979ed/src/framework-configuration.js#L45) Basically the idea is we check for called `configure` fn before calling it